logstash-core/spec/logstash/config/mixin_spec.rb

```diff
@@ -102,6 +102,58 @@
     end
   end
 
+  context "when validating :uri" do
+    let(:klass) do
+      Class.new(LogStash::Filters::Base)  do
+        config_name "fakeuri"
+        config :uri, :validate => :uri
+      end
+    end
+    
+    let(:scheme) { "myscheme" }
+    let(:user) { "myuser" }
+    let(:password) { "fancypants" }
+    let(:hostname) { "myhostname" }
+    let(:path) { "/my/path" }
+    let(:uri_str) { "#{scheme}://#{user}:#{password}@#{hostname}#{path}" }
```

Comment:
Can we add another test -- URI with no username/password to validate if the behavior is the same?